api/v1alpha1/helpers.go

```diff
@@ -17,11 +17,24 @@ const (
 	// GatewayClassControllerRef identifies contour operator as the managing controller
 	// of a GatewayClass.
 	GatewayClassControllerRef = "projectcontour.io/contour-operator"
+
 	// GatewayClassParamsRefGroup identifies contour operator as the group name of a
 	// GatewayClass.
 	GatewayClassParamsRefGroup = "operator.projectcontour.io"
+
 	// GatewayClassParamsRefKind identifies Contour as the kind name of a GatewayClass.
 	GatewayClassParamsRefKind = "Contour"
+
+	// GatewayFinalizer is the name of the finalizer used for a Gateway.
+	GatewayFinalizer = "gateway.networking.x-k8s.io/finalizer"
+
+	// OwningGatewayNameLabel is the owner reference label used for a Gateway
+	// managed by the operator. The value should be the name of the Gateway.
+	OwningGatewayNameLabel = "contour.operator.projectcontour.io/owning-gateway-name"
```

Comment:
I wanted to provide consistency with the [contour owning ns/name labels](https://github.com/projectcontour/contour-operator/blob/main/api/v1alpha1/contour_types.go#L23-L29). Child resources of the Gateway, i.e. Envoy daemonset, would look like:
```
kind: DaemonSet
metadata:
  name: envoy
  namespace: foo-gateway-ns
  labels:
    contour.operator.projectcontour.io/owning-gateway-name: foo-gateway
    contour.operator.projectcontour.io/owning-gateway-namespace: foo-gateway-ns
```

We could drop `contour.operator.projectcontour.io/owning-gateway-namespace` since the child resources of a gateway will always be in the same ns as the gateway. I can see loosing the `contour` prefix, but the Contour owning labels should be updated too for consistency. I'd prefer doing that in a follow-up PR. xref: https://github.com/projectcontour/contour-operator/issues/216